Slice all non significant zeroes in stringifyBalance

brad-decker opened the pull request:
In the MetaMask Extension, having a balance of tokens that would result in a decimal format including significant numbers past the hundredths place would result in a zero balance being displayed for both the token and for fiat currency. Instead of arbitrarily returning 3 decimals of significance, this update uses a regex to trim all zeroes between the decimal point and the end of the string that are uninterrupted by any other value. It will start at the first 0 in a series of zeroes and cut until the end of the string.

'0.000000010000' -> '0.00000001'
'0.000000000000' -> '0'
'0.010000000000' -> '0.01'
'15.00000000000' -> '15'

etc

If the result of substring'ing the fractional part of the number is not a sequence of zeroes, the original logic is preserved:
'15.001981241243' -> '15.001'
'0.0187777777777' -> '0.018'

<details>
<summary>Before</summary>

<img src="https://user-images.githubusercontent.com/4448075/88091589-e2bdc380-cb54-11ea-8873-33b0b8b6b9ce.png"/>

</details>

<details>
<summary>After</summary>

<img width="360" alt="Screen Shot 2020-07-21 at 1 14 54 PM" src="https://user-images.githubusercontent.com/4448075/88091623-f0734900-cb54-11ea-997a-5e503fdb65bf.png">

</details>

brad-decker commented on test/integration/token-precision.js:
```diff
@@ -66,9 +66,9 @@ generateTestWithParams({
   quantity: '2179663820576',
   precision: 18,
-  result: function (token, t) {
+  result: function (token, t) {;
```
oops, will remove this semicolon

rachelcope commented:
This seems like a good solution. Do we currently have a max number of decimals? What happens in the case that number has more than 8 decimals (e.g., 0.000000000000000001)? Could we establish a max and use ellipses in cases the exceed the max (e.g., 0.00...) and the full number is displayed on hover?

brad-decker commented:
@rachelcope yeah, I was planning on doing that in a new PR in the extension- right now though the token tracker returns both a balance and a string representing the balance. Another possible solution that would make this PR void is to use the balance returned here to stringify the balance on our end the way we want. The string returned from the token tracker before this change would be '0.000' for a balance of '0.0001' -- which means I can't display a non truncated version using that value.

brad-decker commented:
FWIW I could also put this behavior behind a configuration option so the new output is opt-in.

brad-decker commented:
I'm fine with a major bump, might be safer for any consumers of the API? Should changelog entries be a part of this PR and version bump a new PR? Or are they each separate PRs?

Gudahtt commented:
A separate PR would make sense; that would be in accordance with [the new release / publishing procedure](https://github.com/MetaMask/controllers#release--publishing), which was proposed for use with all of our libraries.

brad-decker commented:
Cool. Thanks for the link! I updated one line -- at the end of the utility function, I repeated the substr call to get the `whole` variable instead of reusing that variable. Updated it to use `whole` there too.
